src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,123 @@
+package wolox.training.controllers;
+
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.UserIdMismatchException;
+import wolox.training.exceptions.UserNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.Users;
+import wolox.training.repositories.UserRepository;
+
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @GetMapping
+    public List<Users> findAll() {
+        return userRepository.findAll();
+    }
+
+    /**
+     *
+     * @param userName Name of a {@link Users}
+     * @return {@link Users} with the name passed as parameter
+     */
+    @GetMapping("/name/{userName}")
+    public Optional<Users> findByName(@PathVariable String userName) {
```

Comment:
I think that this method should not return optional. 
I believe that the best option should be return not found exception in case the user is not found.
What do you think?

Reply:
Yes you're right. Thanks!!